lib/web_ui/lib/src/engine/pointer_binding.dart

```diff
@@ -1133,12 +1151,21 @@ class _PointerAdapter extends _BaseAdapter with _WheelEventListenerMixin {
   }
 
   int _getPointerId(DomPointerEvent event) {
-    // We force `device: _mouseDeviceId` on mouse pointers because Wheel events
-    // might come before any PointerEvents, and since wheel events don't contain
-    // pointerId we always assign `device: _mouseDeviceId` to them.
-    final ui.PointerDeviceKind kind = _pointerTypeToDeviceKind(event.pointerType!);
-    return kind == ui.PointerDeviceKind.mouse ? _mouseDeviceId :
-        event.pointerId!.toInt();
+    // All mouse pointer events are given `_mouseDeviceId`, including wheel
+    // events, because wheel events might come before any other PointerEvents,
+    // and wheel PointerEvents don't contain pointerIds.
+    return switch(_pointerTypeToDeviceKind(event.pointerType!)) {
+      ui.PointerDeviceKind.mouse => _mouseDeviceId,
+
+      ui.PointerDeviceKind.stylus ||
+      ui.PointerDeviceKind.invertedStylus => _stylusDeviceId,
+
+      // Trackpad processing doesn't call this function.
+      ui.PointerDeviceKind.trackpad => throw Exception('Unreachable'),
```

Comment:
Are we sure of this? What happens if I touch with several fingers on a mac touchpad? Does it get converted to a "touch" event?

Reply:
I've tested on mac Chrome and multitouching with trackpad works. I think they're still converted to "touch" device. Apps probably won't even correctly process touch events with a "trackpad" device. Also this method is just not called by listeners to touchpad.

I could have changed this to the default, but when I read the code I thought it should be ok, and I'd rather change my assumption upon reports than leaving an unreachable branch here. :)

Reply:
Thanks for testing this!